src/unibit.js

```diff
@@ -367,6 +368,15 @@ module.exports = class Unibit {
         return _.get(context.site.data, path, null);
     }
 
+    getPageByReference(context, reference) {
+        // remove extension
+        reference = reference.replace(/\.md$/, '');
+        return _.find(context.site.pages, page => {
+            const pageUrl = _.trim(_.get(page, 'url'), '/');
+            return reference === pageUrl;
+        });
+    }
+
```

Comment:
Every page object should have `relPath` property with a path to the page relative to `pagesDir`.
Use that instead, because `page.url` might actually be different from `page.relPath` if some kind of permalink is used that places the page under a different URL than the page file.

pretty URLs:
`/content/about/index.md` => `/content/about`

permalinks:
`/content/hello.md` => `/content/foo`
```
---
title: Hello
permalink: /foo
---
```